src/python/module/smartredis/dataset_utils.py

```diff
@@ -0,0 +1,141 @@
+        fieldnames and appropriately formatted metadata. 
+        rtype: dict
+        """
+        typecheck(dataset, "dataset", Dataset)
+        
+        coord_dict= {}
+        coord_final = {}
+        variable_names = dataset.get_meta_strings("_xarray_data_names")
+             
+        for j,d in list(permutations(variable_names,2)):
+            for k in dataset.get_meta_strings(f"_xarray_{j}_coord_names")[0].split(","):
+                if d == k:
+                    # Remove coordinate data names from data names - may refactor this to make more clear
+                    variable_names.remove(d)
+                    # Get coordinate dimensions in the appropriate format for Xarray 
+                    coord_dims = []
+                    for coord_dim_field_name in dataset.get_meta_strings(f"_xarray_{d}_dim_names")[0].split(","):
+                        coord_dims.append(dataset.get_meta_strings(coord_dim_field_name)[0])
+                    # Get coordinate attributes in the appropriate format for Xarray 
+                    coord_attrs = {}
+                    for coord_attr_field_name in dataset.get_meta_strings(f"_xarray_{d}_attr_names")[0].split(","):
+                        coord_attrs[coord_attr_field_name] = dataset.get_meta_strings(coord_attr_field_name)[0] 
+                    # Add dimensions, data, and attributes to the coordinate variable       
+                    coord_dict[d]= (coord_dims,dataset.get_tensor(d),coord_attrs)
+                    # Add coordinate names and relative values in the appropriate form to add to Xarray coords variable
+                    coord_final[j] = coord_dict        
+        
+        ret_xarray = {}
+        # Loop thorugh the rest of the data variable and construct xr.DataArrays    
+        for variable_name in variable_names: # for variable_name in variable_names
+            data_final = dataset.get_tensor(variable_name) 
+            dims_final=[]
+            # Extract dimensions in correct form
+            for dim_field_name in dataset.get_meta_strings(f"_xarray_{variable_name}_dim_names")[0].split(","):
+                dims_final.append(dataset.get_meta_strings(dim_field_name)[0])
+            attrs_final = {}
+            #Extract attributes in correct form 
+            for attr_field_name in dataset.get_meta_strings(f"_xarray_{variable_name}_attr_names")[0].split(","):
+                attrs_final[attr_field_name] = dataset.get_meta_strings(attr_field_name)[0]     
+            # Add coordinates to the correct data name
+            for name in coord_final.keys():
+                if name == variable_name: 
+                    coords_final = coord_final.get(name)
+            
+            # Construct a xr.DataArray using extracted dataset data, append the dataarray to corresponding variable names
+            ret_xarray[variable_name] = xr.DataArray(data=data_final,coords=coords_final,dims=dims_final,attrs=attrs_final)
```

Comment:
Do these line lengths exceed pep8 standards?

Reply:
They probably will when spaces are added after the commas, so they need to be split